R/deprecate.R

```diff
@@ -0,0 +1,227 @@
+#' Internal comment functions
+#'
+#' Users are advised to use [wb_add_comment()] and [wb_remove_comment()].
+#' `write_comment()` and `remove_comment()` are now deprecated. openxlsx2 will stop
+#' exporting it at some point in the future. Use the replacement functions.
+#' @name comment_internal
+NULL
+
+#' @rdname comment_internal
+#' @inheritParams wb_add_comment
+#' @param comment An object created by [create_comment()]
+#' @param row,col Row and column of the cell
+#' @param color optional background color
+#' @param file optional background image (file extension must be png or jpeg)
+#' @keywords internal
+#' @export
+#' @inherit wb_add_comment examples
+write_comment <- function(
+    wb,
+    sheet,
+    col     = NULL,
+    row     = NULL,
+    comment,
+    dims    = rowcol_to_dim(row, col),
+    color   = NULL,
+    file    = NULL
+) {
+  .Deprecated("wb_add_comment()", package = "openxlsx2", old = "write_comment()")
+  do_write_comment(
+    wb,
+    sheet,
+    col,
+    row,
+    comment,
+    dims,
+    color,
+    file
+  )
+}
+
+#' @rdname comment_internal
+#' @param gridExpand If `TRUE`, all data in rectangle min(rows):max(rows) X min(cols):max(cols)
+#' will be removed.
+#' @keywords internal
+#' @export
+remove_comment <- function(
+    wb,
+    sheet,
+    col        = NULL,
+    row        = NULL,
+    gridExpand = TRUE,
+    dims       = NULL
+) {
+  .Deprecated("wb_remove_comment()", package = "openxlsx2", old = "remove_comment()")
+  do_remove_comment(
+    wb,
+    sheet,
+    col,
+    row,
+    gridExpand,
+    dims
+  )
+}
+
+#' Convert to Excel data
+#'
+#' Use [convert_to_excel_date()].
+#' @usage NULL
+#' @inheritParams convert_to_excel_date
+#' @keywords internal
+#' @export
+convertToExcelDate <- function(df, date1904 = FALSE) {
+  .Deprecated(old = "convertToExcelDate", new = "convert_to_excel_date", package = "openxlsx2")
```

Comment:
guess we can remove this line? The code will stop on the next either way. Should we still export this? I would simply no longer export this and drop the function after a while.

Reply:
If you stop exporting. People using it will get function not found, which is basically equivalent to dropping it fully (which I don't mind doing) I couldn't find any usage on GitHub.